.github/workflows/release.yml

```diff
@@ -0,0 +1,152 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build And Publish Release
+
+on:
+  release:
+    types:
+      - published
+
+permissions:
+  contents: read
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+jobs:
+  push:
+    permissions:
+      contents: read
+      id-token: write # this is important, it's how we authenticate with Vault
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - linux
+        arch:
+          - amd64
+          - arm64
+    steps:
+      - name: Prepare Matrix Instance
+        run: |
+          echo "PLATFORM_PAIR=${{ matrix.os }}-${{ matrix.arch }}"
+          echo "PLATFORM_PAIR=${{ matrix.os }}-${{ matrix.arch }}" >> $GITHUB_ENV
+          echo "GOOS=${{ matrix.os }}"
+          echo "GOOS=${{ matrix.os }}"     >> $GITHUB_ENV
+          echo "GOARCH=${{ matrix.arch }}"
+          echo "GOARCH=${{ matrix.arch }}" >> $GITHUB_ENV
+          echo "ARCH=${{ matrix.arch }}"
+          echo "ARCH=${{ matrix.arch }}"   >> $GITHUB_ENV
+
+      - name: Retrieve Docker Hub Credentials From Vault
+        uses: rancher-eio/read-vault-secrets@main
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKERHUB_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKERHUB_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ env.DOCKERHUB_USERNAME }}
+          password: ${{ env.DOCKERHUB_PASSWORD }}
+
+      - name: Get Sources
+        uses: actions/checkout@v4
+
+      - name: Docker Meta Data Setup
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.IMAGE }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+          cache: false
+
+      - name: Build and stage
+        run: |
+          ./scripts/build
+          # Stage binary for packaging step
+          cp -r ./bin/* ./package/
+
+      - name: Build image and push by digest
+        id: build
+        uses: docker/build-push-action@v5
+        with:
+          context: package
+          file: package/Dockerfile
+          platforms: ${{ matrix.os }}/${{ matrix.arch }}
+          labels: ${{ steps.meta.outputs.labels }}
+          outputs: type=image,name=${{ env.IMAGE }},push-by-digest=true,name-canonical=true,push=true
```

Comment:
The link https://github.com/enrichman/cli/actions/runs/8754884403/job/24027726888 points to a good run of build.

>  I tagged my fork to trigger

Fork means you used an actual fork of the cli repo for your PR ?
That does not work with the Vault the creds come from.
In my EIO issue (https://github.com/rancherlabs/eio/issues/2165) I was told clearly that 
> It won't work if you create a PR from a fork

Which is why my own testing at https://github.com/rancher/kube-api-auth/pull/25 was done with a PR from a non-fork. And https://github.com/rancher/kube-api-auth/pull/24 demonstrated the non-working mode from a fork PR.

I am not sure about the `push`. If you go to my test PR it did push results to dockerhub, with the flow as is.
(Results at https://hub.docker.com/layers/rancher/kube-api-auth/pr-25/images/sha256-9f13ef730a7683a7eb91f1258d0129d5000a43c762605e234aa142d8f2a6d2fc?context=explore)

Not convinced that I have to change things here.

Reply:
> The link https://github.com/enrichman/cli/actions/runs/8754884403/job/24027726888 points to a good run of build.

The build is working, but it didn't push anything. If you check the `Docker Build` step output it will show the `push: false` value:

```
Run docker/build-push-action@v5
  with:
    context: package
    tags: enrichman/cli[2](https://github.com/enrichman/cli/actions/runs/8754884403/job/24027726888#step:11:2):v2.9.0-rc1
    load: false
    no-cache: false
    pull: false
    push: false
    github-token: ***
```

> > I tagged my fork to trigger
> 
> Fork means you used an actual fork of the cli repo for your PR ? That does not work with the Vault the creds come from. In my EIO issue ([rancherlabs/eio#2165](https://github.com/rancherlabs/eio/issues/2165)) I was told clearly that
> 
> > It won't work if you create a PR from a fork
> 
> Which is why my own testing at #25 was done with a PR from a non-fork. And #24 demonstrated the non-working mode from a fork PR.
> 
> I am not sure about the `push`. If you go to my test PR it did push results to dockerhub, with the flow as is. (Results at https://hub.docker.com/layers/rancher/kube-api-auth/pr-25/images/sha256-9f13ef730a7683a7eb91f1258d0129d5000a43c762605e234aa142d8f2a6d2fc?context=explore)
> 
> Not convinced that I have to change things here.

Ah nice. Yes, I was going to use a tag on my fork, and my credentials, because EIO haven't migrated the secrets yet. So if it worked it's good to me.

As you pointed out yes, it was because of the ouputs, I missed that!
https://docs.docker.com/reference/cli/docker/buildx/build/#image